homeassistant/components/solarlog/config_flow.py

```diff
@@ -86,21 +91,31 @@ async def async_step_user(self, user_input=None):
                     vol.Required(
                         CONF_HOST, default=user_input.get(CONF_HOST, DEFAULT_HOST)
                     ): str,
+                    vol.Required("extended_data", default=False): bool,
                 }
             ),
             errors=self._errors,
         )
 
     async def async_step_import(self, user_input=None):
         """Import a config entry."""
-        host_entry = user_input.get(CONF_HOST, DEFAULT_HOST)
 
-        url = urlparse(host_entry, "http")
-        netloc = url.netloc or url.path
-        path = url.path if url.netloc else ""
-        url = ParseResult("http", netloc, path, *url[3:])
-        host = url.geturl()
+        if CONF_HOST in user_input:
+            url = urlparse(user_input[CONF_HOST], "http")
+            netloc = url.netloc or url.path
+            path = url.path if url.netloc else ""
+            url = ParseResult("http", netloc, path, *url[3:])
+            user_input[CONF_HOST] = url.geturl()
+
+            if self._host_in_configuration_exists(user_input[CONF_HOST]):
+                return self.async_abort(reason="already_configured")
+        else:
+            user_input |= {CONF_HOST: DEFAULT_HOST}
+
+        if CONF_NAME not in user_input:
+            user_input |= {CONF_NAME: DEFAULT_NAME}
+
+        if "extended_data" not in user_input:
+            user_input |= {"extended_data": False}
```

Comment:
```suggestion
        user_input = {
            CONF_HOST: DEFAULT_HOST,
            CONF_NAME: DEFAULT_NAME,
            "extended_data": False,
            **user_input,
       }
```

Validation is done in user step

Reply:
Took this over with as less changes as possible. But your simplification is much appreciated!

Reply:
I noticed that with the above change, the result of the test against double entries has changed: Before, in `async_step_import` the setup was aborted if the entry already exists (`return self.async_abort(reason="already_configured")`), now with the test only being performed in `async_step_user`, the user is shown the form to change the host name. I therefore added the host validation again with https://github.com/home-assistant/core/pull/117484/commits/8aa26bccd6f4f07083705002e93258f4b7801de5.

Reply:
Can you please create a private function to avoid duplicate code?

Reply:
done